File: Day48/HammingDistance.py
class Solution(object):
    def hammingDistance(self, x, y):
        xor = x ^ y
        count = 0
        while xor:
            xor &= xor - 1
            count += 1
        return count

if __name__ == "__main__":
    x = int(input())
    y = int(input())
    sol = Solution()
    print(sol.hammingDistance(x, y))

File: Day48/HammingDistance.java
import java.util.*;

public class HammingDistance {
    public int hammingDistance(int x, int y) {
        int xor = x ^ y, count = 0;
        while (xor != 0) {
            xor &= xor - 1;
            count++;
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int x = sc.nextInt();
        int y = sc.nextInt();
        HammingDistance sol = new HammingDistance();
        System.out.println(sol.hammingDistance(x, y));
    }
}
